plugins/outputs/stackdriver/stackdriver.go

```diff
@@ -70,17 +76,31 @@ func (s *Stackdriver) Connect() error {
 	return nil
 }
 
+// sorted returns a copy of the metrics in time ascending order
+func sorted(metrics []telegraf.Metric) []telegraf.Metric {
+	batch := make([]telegraf.Metric, 0, len(metrics))
+	for i := len(metrics) - 1; i >= 0; i-- {
+		batch = append(batch, metrics[i])
+	}
+	sort.Slice(batch, func(i, j int) bool {
```

Comment:
Why not just `sort.Slice` `metrics`, instead of duplicating it into `batch`?

Reply:
That's a good question, the reason is that outputs are not allowed to modify the batch because if an error occurs the metrics will not be restored to the metric buffer correctly.  This is an unfortunate aspect of the output interface, it does save a copy of the buffer but it might be worth the copy to prevent misuse.